tsdb/chunks/head_chunks_test.go

```diff
@@ -246,29 +233,35 @@ func TestHeadReadWriter_Truncate(t *testing.T) {
 	testutil.Ok(t, hrw.IterateAllChunks(func(_, _ uint64, _, _ int64, _ uint16) error { return nil }))
 	testutil.Assert(t, hrw.fileMaxtSet, "")
 
-	// Truncating files after restart. As the last file was empty, this creates no new files.
-	testutil.Ok(t, hrw.Truncate(timeToTruncateAfterRestart))
-	verifyFiles(totalFiles-filesDeletedAfter2ndTruncation, startIndexAfter2ndTruncation)
+	verifyFiles([]int{3, 4, 5, 6, 7, 8})
+	// New file is created after restart even if last file was empty.
+	addChunk()
+	verifyFiles([]int{3, 4, 5, 6, 7, 8, 9})
+
+	// Truncating files after restart.
+	testutil.Ok(t, hrw.Truncate(sixthFileMint))
+	verifyFiles([]int{6, 7, 8, 9, 10})
 
-	// First chunk after restart creates a new file.
+	// As the last file was empty, this creates no new files.
+	testutil.Ok(t, hrw.Truncate(sixthFileMint+1))
+	verifyFiles([]int{6, 7, 8, 9, 10})
 	addChunk()
-	totalFiles++
 
 	// Truncating till current time should not delete the current active file.
-	testutil.Ok(t, hrw.Truncate(int64(timeRange+fileTimeStep)))
-	verifyFiles(2, totalFiles) // One file is the active file and one was newly created.
+	testutil.Ok(t, hrw.Truncate(int64(timeRange+(2*fileTimeStep))))
+	verifyFiles([]int{10, 11}) // One file is the previously active file and one was newly created.
 }
 
-// TestHeadReadWriter_Truncate_NoUnsequentialFiles tests
+// TestChunkDiskMapper_Truncate_NoUnsequentialFiles tests
 // that truncation leaves no unsequential files on disk, mainly under the following case
 // * There is an empty file in between the sequence while the truncation
 //   deletes files only upto a sequence before that (i.e. stops deleting
 //   after it has found a file that is not deletable).
 // This tests https://github.com/prometheus/prometheus/issues/7412 where
 // the truncation used to check all the files for deletion and end up
 // deleting empty files in between and breaking the sequence.
-func TestHeadReadWriter_Truncate_NoUnsequentialFiles(t *testing.T) {
-	hrw, close := testHeadReadWriter(t)
+func TestChunkDiskMapper_Truncate_NoUnsequentialFiles(t *testing.T) {
```

Comment:
`NoUsequentialFiles` is a double negation, which is difficult to understand. How about `PreservesFileSequence`?

I would also like to suggest a more straightforward doc comment:

TestChunkDiskMapper_Truncate_PreservesFileSequence tests that truncation doesn't poke holes into the file sequence, even if there are empty files in between non-empty files. This test exposes https://github.com/prometheus/prometheus/issues/7412 where the truncation simply deleted all empty files instead of stopping once it encountered a non-empty file.